api/storage_service.cc

```diff
@@ -900,6 +902,129 @@ void set_storage_service(http_context& ctx, routes& r) {
+                        }
+                    }
+                }
+            }, [ks, cf](const database& db) {
+                table_sstables_list res;
+
+                auto& ext = db.get_config().extensions();
+
+                for (auto& t : db.get_column_families() | boost::adaptors::map_values) {
+                    auto& schema = t->schema();
+                    if ((ks.empty() || ks == schema->ks_name()) && (cf.empty() || cf == schema->cf_name())) {
+                        ss::table_sstables tst;
+                        tst.keyspace = schema->ks_name();
+                        tst.table = schema->cf_name();
+
+                        for (auto sstable : *t->get_sstables_including_compacted_undeleted()) {
+                            auto ts = db_clock::to_time_t(sstable->data_file_write_time());
+                            ::tm t;
+                            ::localtime_r(&ts, &t);
+
+                            ss::sstable info;
+
+                            info.timestamp = t;
+                            info.generation = sstable->generation();
+                            info.level = sstable->get_sstable_level();
+                            info.size = sstable->bytes_on_disk();
+                            info.data_size = sstable->ondisk_data_size();
+                            info.index_size = sstable->index_size();
+                            info.filter_size = sstable->filter_size();
+                            info.version = sstable->get_version();
+
+                            if (sstable->has_component(sstables::component_type::CompressionInfo)) {
+                                auto& c = sstable->get_compression();
+                                auto cp = sstables::get_sstable_compressor(c);
+
+                                ss::named_maps nm;
+                                nm.group = "compression_parameters";
+                                for (auto& p : cp->options()) {
+                                    ss::map_entry e;
+                                    e.key = p.first;
+                                    e.value = p.second;
+                                    nm.attributes.push(std::move(e));
+                                }
+                                if (!cp->options().count(compression_parameters::SSTABLE_COMPRESSION)) {
+                                    ss::map_entry e;
+                                    e.key = compression_parameters::SSTABLE_COMPRESSION;
+                                    e.value = cp->name();
+                                    nm.attributes.push(std::move(e));
+                                }
+                                info.extended_properties.push(std::move(nm));
+                            }
+
+                            sstables::file_io_extension::attr_value_map map;
+
+                            for (auto* ep : ext.sstable_file_io_extensions()) {
+                                ep->get_attributes(*sstable, map);
```

Comment:
I see now why (the map is not returned). Well there is `std::map::merge()` now but it will be slightly less efficient than this (keys will be looked up twice). IMO the code would be more natural, but no strong opinion.

Reply:
I sort of agree, but it would also (for better or worse) not allow for example overriding an existing attribute (since merge does not). Not sure if there is any market for this particular feature really, but I sort of wanted it just in case. Of course it also means extensions have potentially to much control, which is a problem in the other direction.

Reply:
Actually, I'm gonna go with your objection. I've changed the API. It is cleaner. New version pushed.